File: backend/src/main/java/com/soen345/ticketreservation/model/Reservation.java
package com.soen345.ticketreservation.model;

import com.soen345.ticketreservation.model.enums.ReservationStatus;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reservations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_id", nullable = false)
    private Event event;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ReservationStatus status = ReservationStatus.CONFIRMED;

    @Column(nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime reservedAt = LocalDateTime.now();

    @Column(nullable = false, unique = true)
    private String confirmationCode;
}

File: app/src/main/java/com/example/frontend/model/Reservation.kt
package com.example.frontend.model

enum class ReservationStatus {
    CONFIRMED, CANCELLED
}

data class ReservationRequest(
    val eventId: Long,
    val numberOfTickets: Int = 1
)

data class ReservationResponse(
    val id: Long,
    val userId: Long,
    val userName: String,
    val eventId: Long,
    val eventTitle: String,
    val status: ReservationStatus,
    val reservedAt: String,
    val confirmationCode: String,
    val numberOfTickets: Int
)
